aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -489,11 +506,8 @@ public void clear(@NonNull Action onComplete,
 
     private void beforeOperation(@NonNull final Runnable runnable) {
         try {
-            Completable.fromAction(
-                () -> {
-                    categoryInitializationsPending.await();
-                    orchestrator.start();
-                })
+            categoryInitializationsPending.await();
+            orchestrator.start(BEFORE_OP_TIMEOUT_MS, TimeUnit.MILLISECONDS)
```

Comment:
I'm a little confused by this. What does it mean to have a 0 second timeout? Would it be better to just fork the `start()` API, and have `startWithBehaviorFoo()` and `startWithBehaviorBar()`?

Reply:
Man I was confused when I first saw this, then saw his explanation and understood it, but now I'm confused again. If it's zero second timeout, does it not _ALWAYS_ fail? and return `false` regardless of whether transition is already happening or not? Maybe it should be a small value like 200ms instead?

Reply:
Timeout of 0 means that the caller only wants the lock if one is available immediately. Basically a fail fast. That being said, I can achieve similar behavior after refactoring the way the completable is built inside the orchestrator.

Reply:
I removed the new overload with the timeout parameter. Just had to re-work the completable a bit.